subsys/llext/llext.c

```diff
@@ -23,12 +23,32 @@ static sys_slist_t _llext_list = SYS_SLIST_STATIC_INIT(&_llext_list);
 
 static struct k_mutex llext_lock = Z_MUTEX_INITIALIZER(llext_lock);
 
-ssize_t llext_find_section(struct llext_loader *ldr, const char *search_name)
+ssize_t llext_section_offset(struct llext_loader *ldr, struct llext *ext, const char *search_name)
```

Comment:
The `find_section` API was IMO terribly broken, since it does not tell you the size associated with the section; it is unclear to me how to safely use it. :man_shrugging: 
Can we take this opportunity to provide a replacement where you get the full section header information instead? SOF can just refer to the `sh_offset` field in the returned struct.

```suggestion
const elf_shdr_t *llext_get_section_header(struct llext_loader *ldr, struct llext *ext, const char *search_name)
```